pkg/cli/build.go

```diff
@@ -227,13 +253,101 @@ func buildCmd() *cobra.Command {
 	cmd.Flags().StringSliceVar(&lintWarn, "lint-warn", linter.DefaultWarnLinters(), "linters that will generate warnings")
 	cmd.Flags().BoolVar(&ignoreSignatures, "ignore-signatures", false, "ignore repository signature verification")
 	cmd.Flags().BoolVar(&cleanup, "cleanup", true, "when enabled, the temp dir used for the guest will be cleaned up after completion")
+	cmd.Flags().StringVar(&configFileGitCommit, "git-commit", "", "commit hash of the git repository containing the build config file (defaults to detecting HEAD)")
+	cmd.Flags().StringVar(&configFileGitRepoURL, "git-repo-url", "", "URL of the git repository containing the build config file (defaults to detecting from configured git remotes)")
+	cmd.Flags().StringVar(&configFileLicense, "license", "NOASSERTION", "license to use for the build config file itself")
 
 	_ = cmd.Flags().Bool("fail-on-lint-warning", false, "DEPRECATED: DO NOT USE")
 	_ = cmd.Flags().MarkDeprecated("fail-on-lint-warning", "use --lint-require and --lint-warn instead")
 
 	return cmd
 }
 
+type gitState struct {
+	// The current git commit, from which we're building packages. e.g. "0c1f2c1f3fe13e8b01f4fccbc5880525739d74a2".
+	commit string
+
+	// e.g. "github.com" (as in from "github.com/wolfi-dev/os")
+	repoHost string
+
+	// e.g. "wolfi-dev"
+	repoPathParent string
+
+	// e.g. "os"
+	repoName string
```

Comment:
I also need committimestamp in unixtime() please to be used as builddate & source-date-epoch. was working on adding this in another pr. But also easy to extend this later to do that too.

Reply:
Cool, yeah let's make sure to unify how we do this if it's diverging — in fact this PR is already cleanup up duplicate implementations of this kind of commit logic. We'll stay in sync!